sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala

```diff
@@ -248,6 +249,10 @@ private[hive] class HiveMetastoreCatalog(sparkSession: SparkSession) extends Log
           if (inferenceMode == INFER_AND_SAVE) {
             updateCatalogSchema(relation.tableMeta.identifier, schema)
           }
+
+          SchemaUtils.checkSchemaColumnNameDuplication(
```

Comment:
I checked code again and it seems I found no column duplication happen here if the hive catalog has no name duplication. `mergeWithMetastoreSchema` merges a catalog schema with an inferred one on the catalog names, so IIUC this does not lead to the duplication. So, we could safely remove this check?